### PR TITLE
feat(examples): in-memory search

### PR DESCRIPTION
**Pull request by Chriscbr**
A random idea I had for a Wing example after reading this blog post: https://boyter.org/posts/abusing-aws-to-make-a-search-engine/

It's a bit non-sensical, but thought I'd share just for fun. Doesn't have to be merged if it seems too out there 🙂 

*By submitting this pull request, I confirm that my contribution is made under the terms of the 
[Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: examples/proposed/in-memory-search.w [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  // - fetch url
+  // - extract title from <title> tag
+  // - extract description from <meta name="description"> tag
+};
+
+let make_worker = (chunk: Map<Array<str>>): cloud.Function => {
+  return new cloud.Function(inflight (terms: Array<str>): Array<str> => {
+    let website_urls = new MutSet<str>();
+    for term in terms {
+      if term in partial_index {
```
[the remark this answers is not in the document]
Comment (Chriscbr) [reply examples/proposed/in-memory-search.w]: Fixed, this should be `chunk`

File: examples/proposed/in-memory-search.w [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        website_urls.add(...partial_index[term]);
```
[the remark this answers is not in the document]
Comment (Chriscbr) [reply examples/proposed/in-memory-search.w]: No prob - I swapped it with a for loop for now

**Comment by Chriscbr** [comment]
@eladb It's a good idea, I guess I just didn't want to make the example too long. 😅

(Better yet, maybe the search index should be produced by a separate `web-scraper.w` example? pull requests welcome 😉)

File: examples/proposed/cacheable-function.w [hunk 3]
```diff
@@ -9,7 +9,7 @@ resource CachableFunction extends cloud.Function {
         this._cache = new cloud.Bucket();
     }
 
-    public override inflight invoke(event: any) {
+    public override inflight invoke(event: any): any {
```
Comment (staycoolcall911) [inline examples/proposed/cacheable-function.w:12]: I think we don't have `any` in wing, but I guess it looks better than those `: str` return types we have everywhere, so maybe just keep it as is (just `bring`ing this point to attention).

**Review by staycoolcall911 — APPROVED** [review]

**Pull request merged**